### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by ahmetbozaci**
API endpoints :

- list all posts.
- list all comments for a post.
- add a comment to a post
- Add authentication to your API endpoints.

### CODE REVIEW

File: app/auth/authenticate_user.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  # verify user credentials
+  def user
+    user = User.find_by(email: email)
+    return user if user&.authenticate(password)
+
+    # raise Authentication error if credentials are invalid
+    raise(ExceptionHandler::AuthenticationError, Message.invalid_credentials)
+  end
+end
```
Comment (Emmyn5600) [inline app/auth/authenticate_user.rb:24]: - [ ] Endpoint of the login or authentication doesn't work as expected. Please fix it.

api4

File: app/controllers/authentication_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    auth_token = AuthenticateUser.new(auth_params[:email], auth_params[:password]).call
+    json_response(auth_token: auth_token)
+  end
+
+  private
+
+  def auth_params
+    params.permit(:email, :password)
+  end
+end
```
Comment (Emmyn5600) [inline app/controllers/authentication_controller.rb]: - [ ] Kindly, return the response in the JSON format after sending a request.

api3

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @ahmetbozaci  @emmanuellekamwa  and @CollinsTatang 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️

- Kindly, include the endpoints you have implemented in the readme file.

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by AshakaE — APPROVED** [review]
### Hi @ahmetbozaci   🙋‍♀️, 

# STATUS APPROVED ✔
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**